### PR TITLE
refactor dns

### PR DESCRIPTION
**Pull request by daixiang0**
Signed-off-by: Long <long0dai@foxmail.com>

Sync from https://github.com/kubeedge/kubeedge/pull/1840

### CODE REVIEW

**Review by Poorunga — COMMENTED** [review]
Should we delete the `agent/pkg/dns/lib.go` file?

File: agent/pkg/dns/dns.go [hunk 1]
```diff
@@ -1,299 +1,57 @@
...
-				klog.Errorf("failed to wirte to udp, err: %v", err)
-				continue
-			}
-			break
-		}
-	}
-}
-
-// parseNameServer gets all real nameservers from the resolv.conf
-func (dns *EdgeDNS) parseNameServer() ([]net.IP, error) {
```
[the remark this answers is not in the document]
Comment (daixiang0) [reply agent/pkg/dns/dns.go:170]: Let's keep it now.

[the remark this answers is not in the document]
Comment (Poorunga) [reply agent/pkg/dns/dns.go:170]: Yes, before we remove the dependency on docker0, we must keep the nameserver 172.17.0.1.

[the remark this answers is not in the document]
Comment (Poorunga) [reply agent/pkg/dns/dns.go:170]: I saw you another pr(https://github.com/kubeedge/edgemesh/pull/36), it’s unbelievable that you finished it so fast, wow!

[the remark this answers is not in the document]
Comment (daixiang0) [reply agent/pkg/dns/dns.go:170]: Fast action and feedback, I love it :)

[the remark this answers is not in the document]
Comment (Poorunga) [reply agent/pkg/dns/dns.go:170]: > I maintained edgemesh for some time actually.

Thank you very much for your contribution, and we sincerely hope that you can continue to participate in the evolution of edgemesh.🤞

**Review by fisherxu — COMMENTED** [review]
/approve

**Pull request merged**